### ISSUE TITLE
the training loss of QHNet did not drop

### ISSUE THREAD

**Issue by EperLuo** [divelab/AIRS#8]
Hi! I was running the QHNet model using the QH9 dataset these days, but the training loss and validation loss did drop after a few steps. The mae for hamiltonian is 0.05485373 at the begining but it only dropped to 0.01 at the 40,000th step, which is quite weird.

I didn't change any of the code in the original codebase and the dataset was directly downloaded from the Google Drive. I run the main.py directly, and I couldn't find the reason why I can't reproduce the experimental results from the paper.

I wonder if you guys have any comments about that. Any help would be much appreciated!

**Comment by Oceanusity** [comment]
Hi, would you mind providing more details about this problem? For example, do you have the log file when training the code? What is the split you have used? Thanks.

**Comment by EperLuo** [comment]
hi! thanks for the reply. I was using the random split of QH9 stable dataset. I've fixed the issue by changing my torch version, it seems like that my torch version is too high. But it still weird to me why the version of package can effect the training loss since the training code went very well.

And I got another question about the model code. When generating spherical harmonics for the edge vectors, you seems switch the order of x, y and z. I wander why you do that, is the pos order in the dataset different from x-y-z?

Any answer would be much appreciated!

**Comment by Oceanusity** [comment]
Would you mind sharing the torch version that cause the problem and the version that works? It help me check the problem. Thanks. 

For the implementation of e3nn, it is a little different from the real spherical harmonics. Please refer to this issue for details (https://github.com/e3nn/e3nn/issues/369). I switch this to make it consistent to the [real spherical harmonics table on Wiki](https://en.wikipedia.org/wiki/Table_of_spherical_harmonics#Real_spherical_harmonics).
It is not sure whether the change may cause or not cause trouble to predict Hamiltonian matrix with predefined magnetic order. Welcome to post information if anyone has idea.

**Comment by EperLuo** [comment]
Thanks for the reply! I was wrong about my torch version, actually I fixed the problem by switching to a higher version. The torch version that cause the problem was 1.13.1, while the version that works well is 2.1.2.
